azuread_service_principal - support for the AppRoleAssignmentRequired property

dmeenhuis opened the pull request:
This adds support for setting the `AppRoleAssignmentRequired` property on a service principal.

I'm not sure if I implemented the update function correctly, but since it was not already there I think I had to create it.

katbyte commented on azuread/resource_service_principal_test.go:
```diff
@@ -131,7 +133,8 @@ resource "azuread_application" "test" {
 resource "azuread_service_principal" "test" {
-  application_id = "${azuread_application.test.application_id}"
+	application_id = "${azuread_application.test.application_id}"
+	app_role_assignment_required = true
```
COuld we align these assingments here?

dmeenhuis replied:
Done

katbyte replied:
`oauth2_permissions` would be great, tags however we always leave at the end. Thanks!

katbyte commented on website/docs/r/service_principal.html.markdown:
```diff
@@ -27,6 +27,7 @@ resource "azuread_application" "example" {
 
 resource "azuread_service_principal" "example" {
   application_id = "${azuread_application.example.application_id}"
+  app_role_assignment_required  = false
```
Could we aling these assingments here?

dmeenhuis replied:
Done

katbyte commented on azuread/resource_service_principal.go:
```diff
@@ -44,6 +45,11 @@ func resourceServicePrincipal() *schema.Resource {
 				Computed: true,
 			},
 
+			"app_role_assignment_required": {
```
Could we move this above the computed properties?

dmeenhuis replied:
Done. What about the `oauth2_permissions` and `tags`, would you like me to move those above the computed properties as well, for consistency's sake?

dmeenhuis commented:
I added a test to check the update function. Wasn't entirely sure how to go about it, so I took the update test from application and followed a similar approach for service principal and the test seems to pass. Could you let me know whether I implemented it correctly?

katbyte commented on website/docs/r/service_principal.html.markdown:
```diff
@@ -36,43 +36,47 @@ resource "azuread_service_principal" "example" {
 The following arguments are supported:
 
-* `application_id` - (Required) The ID of the Azure AD Application for which to create a Service Principal.
+- `application_id` - (Required) The ID of the Azure AD Application for which to create a Service Principal.
```
Could we not change these from `*` to `-` for consistency with the rest of the provider docs?

dmeenhuis commented:
Thanks for merging these changes! I wasn't able to respond earlier since I was on holidays for the past 3 weeks.
